src/http/http2_endpoint.h

```diff
@@ -0,0 +1,269 @@
+        configuration, // Note: Support configuration
+      const std::shared_ptr<ErrorReporter>& error_reporter =
+        nullptr) // Note: Report errors
+      :
+      HTTP2Endpoint(server_session, session_id, writer_factory, std::move(ctx)),
+      server_session(*this, *this),
+      rpc_map(rpc_map),
+      session_id(session_id),
+      interface_id(interface_id)
+    {}
+
+    void send(std::vector<uint8_t>&& data, sockaddr) override
+    {
+      send_raw(std::move(data));
+    }
+
+    void handle_request(
+      llhttp_method verb,
+      const std::string_view& url,
+      http::HeaderMap&& headers,
+      std::vector<uint8_t>&& body,
+      int32_t stream_id) override
+    {
+      LOG_TRACE_FMT(
+        "Processing msg({}, {} [{} bytes])",
+        llhttp_method_name(verb),
+        url,
+        body.size());
+
+      try
+      {
+        if (session_ctx == nullptr)
+        {
+          session_ctx = std::make_shared<ccf::SessionContext>(
+            session_id, peer_cert(), interface_id);
+        }
+
+        std::shared_ptr<http::HttpRpcContext> rpc_ctx = nullptr;
+        try
+        {
+          rpc_ctx = std::make_shared<HttpRpcContext>(
+            session_ctx, verb, url, std::move(headers), std::move(body));
+        }
+        catch (std::exception& e)
+        {
+          // Note: return HTTP_STATUS_INTERNAL_SERVER_ERROR, e.what()
+        }
+
+        const auto actor_opt = http::extract_actor(*rpc_ctx);
+        if (!actor_opt.has_value())
+        {
+          rpc_ctx->set_error(
+            HTTP_STATUS_NOT_FOUND,
+            ccf::errors::ResourceNotFound,
+            fmt::format(
+              "Request path must contain '/[actor]/[method]'. Unable to parse "
+              "'{}'.",
+              rpc_ctx->get_method()));
+          // Note: return HTTP_STATUS_NOT_FOUND, e.what()
+          return;
+        }
+
+        const auto& actor_s = actor_opt.value();
+        auto actor = rpc_map->resolve(actor_s);
+        auto search = rpc_map->find(actor);
+        if (actor == ccf::ActorsType::unknown || !search.has_value())
+        {
+          rpc_ctx->set_error(
+            HTTP_STATUS_NOT_FOUND,
+            ccf::errors::ResourceNotFound,
+            fmt::format("Unknown actor '{}'.", actor_s));
+          // Note: return HTTP_STATUS_NOT_FOUND, e.what()
+          return;
+        }
+
+        auto response = search.value()->process(rpc_ctx);
+
+        if (!response.has_value())
+        {
+          // If the RPC is pending, hold the connection.
+          LOG_TRACE_FMT("Pending");
+          return;
+        }
+        else
+        {
+          server_session.send_response(
+            stream_id,
+            rpc_ctx->get_response_http_status(),
+            rpc_ctx->get_response_headers(),
+            std::move(rpc_ctx->get_response_body()));
+        }
+      }
+      catch (const std::exception& e)
+      {
+        // Note: return HTTP_STATUS_INTERNAL_SERVER_ERROR, e.what()
```

Comment:
```suggestion
```

Reply:
Will be addressed when the two endpoints are merged.